### PR TITLE
ref: remove PR closed duplication

### PR DESCRIPTION
**Pull request by lynnagara**

### CODE REVIEW

File: lib/pr.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from __future__ import annotations
```
Comment (bukzor) [inline lib/pr.py:1]: P4 nit: rename to `lib/gh/pr.py` ?

I want to move most of what's in spec/gh, eventually, pretty sure.

**Review by bukzor — APPROVED** [review]

File: lib/pr.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+@dataclass(frozen=True)
+class PR:
+    url: URL
+
+    def is_closed(self) -> bool:
+        status = sh.stdout(
+            ("gh", "pr", "view", self.url, "--json", "state", "--jq", ".state")
+        )
+        return status in ("CLOSED", "MERGED")
```
Comment (bukzor) [inline lib/pr.py:17]: P3: With just one attribute, this class is better represented as a function:

```suggestion
    def is_closed(pr_url: URL) -> bool:
        status = sh.stdout(
            ("gh", "pr", "view", pr_url, "--json", "state", "--jq", ".state")
        )
        return status in ("CLOSED", "MERGED")
```

File: lib/tf_lock/tf_lock_acquire.py [hunk 3]
```diff
@@ -46,7 +48,7 @@ def tf_lock_acquire(root_module: Path) -> ExitCode:
 
             tf_lock_user = TFLockUser.from_string(lock_user)
             if tf_lock_user.pr_url:  # a pr holds the lock.
-                if is_pr_closed(tf_lock_user.pr_url):
+                if PR(URL(tf_lock_user.pr_url)).is_closed():
```
Comment (bukzor) [inline lib/tf_lock/tf_lock_acquire.py:51]: p3 would be good for `pr_url` to already have type URL.

Comment (lynnagara) [reply lib/tf_lock/tf_lock_acquire.py:51]: i wonder if the URL type is actually needed... it doesn't seem to do much

**Comment by lynnagara** [comment]
they are no longer going to be the same due to https://github.com/getsentry/tacos-gha/pull/251

**Pull request closed without merging**